src/TransitionSystems/location_id.rs

```diff
@@ -7,11 +7,35 @@ pub enum LocationID {
     Conjunction(Box<LocationID>, Box<LocationID>),
     Composition(Box<LocationID>, Box<LocationID>),
     Quotient(Box<LocationID>, Box<LocationID>),
-    Simple(String),
+    Simple(SimpleID),
```

Comment:
Or even better have two different enums one with the component id and one without

Reply:
> Why make `Simple` contain a stuct which have 2 fields instead of making `Simple` hold both the fields? Like all enum did above.

I think that would hurt readability. `Simple(String, String)` tells you nothing about what each `String` does what, Although type alias would fix this, I think this is the best compromise... I will get to work implementing that.

> Or even better have two different enums one with the component id and one without

I don't think this is a great solution:

```
Simple(LocationId, Option(ComponentId))
```

Is much less verbose than

```
NoncompleteSimple(LocatinId)
CompleteSimple(LocationId, ComponentId)
```

Reply:
> Why make `Simple` contain a stuct which have 2 fields instead of making `Simple` hold both the fields? Like all enum did above.

I don't understand what quality this change would add, could you please explain?

Reply:
> ```
> NoncompleteSimple(LocatinId)
> CompleteSimple(LocationId, ComponentId)
> ```

I think that would work nicely

Reply:
Just an FYI
> I think that would hurt readability. Simple(String, String) tells you nothing about what each String does
You can name the fields in an `enum` https://doc.rust-lang.org/rust-by-example/custom_types/enum.html
```rust
Simple {location_id: String, component_id: String}
```
Idk if this helps or anything

Reply:
For sure 👌